frontend/src/components/pages/RegistrantExperience/index.tsx

```diff
@@ -13,10 +21,78 @@ enum RegistrantExperienceSteps {
   ReviewRegistrationPage,
 }
 
+const waiverReducer = (waiverInterface: WaiverInterface, action: any) => {
+  switch (action.type) {
+    case WaiverActions.LOADED_WAIVER: {
+      const optionalClauses: OptionalClauseResponse[] = [];
+      const requiredClauses: RequiredClauseResponse[] = [];
+      const { waiver } = action.payload;
+
+      waiver.clauses.forEach((clause: WaiverClause) => {
+        if (clause.required) optionalClauses.push({ ...clause, agreed: false });
+        else requiredClauses.push(clause);
+      });
+      return {
+        ...waiverInterface,
+        optionalClauses,
+        requiredClauses,
+        waiver,
+        agreedRequiredClauses: false,
+        loadingWaiver: false,
+      };
+    }
+    case WaiverActions.ClICK_REQUIRED_CLAUSE:
+      return {
+        ...waiverInterface,
+        agreedRequiredClauses: !waiverInterface.agreedRequiredClauses,
+      };
+
+    case WaiverActions.CLICK_OPTIONAL_CLAUSE: {
+      if (!waiverInterface.optionalClauses) return waiverInterface;
+      const changedClause: OptionalClauseResponse = action.payload;
+
+      const newOptionalClauses:
+        | OptionalClauseResponse[]
```

Comment:
what's the `|` operator before `OptionalClauseResponse[]` do?

Reply:
I don't think the `|` operator does anything. I believe `yarn fix` places it there for styling.